delphi/LeapfrogInterface.pas

```diff
@@ -0,0 +1,955 @@
+    procedure SetCtxValIsPercent(var inCtxValIsPercent: TGBFixedArray<Integer>);
+    procedure SetHcArtIsAgeSpec(var inHcArtIsAgeSpec: TGBFixedArray<Integer>);
+    procedure SetHcAgeCoarse(var inHcAgeCoarse: TGBFixedArray<Double>);
+    procedure SetAbortion(var inAbortion: TGBFixedArray<Double>);
+    procedure SetPatientsReallocated(var inPatientsReallocated: TGBFixedArray<Double>);
+    procedure SetHcArtLtfu(var inHcArtLtfu: TGBFixedArray<Double>);
+    procedure SetHcAgeCoarseCd4(var inHcAgeCoarseCd4: TGBFixedArray<Integer>);
+    procedure SetAdultFemaleInfections(var inAdultFemaleInfections: TGBFixedArray<Double>);
+    procedure SetAdultFemaleHivnpop(var inAdultFemaleHivnpop: TGBFixedArray<Double>);
+    procedure SetTotalBirths(var inTotalBirths: TGBFixedArray<Double>);
+    procedure SetCtxEffect(var inCtxEffect: TGBFixedArray<Double>);
+    procedure SetHcArtStart(var inHcArtStart: TGBFixedArray<Double>);
+    procedure SetLocalAdjFactor(var inLocalAdjFactor: TGBFixedArray<Double>);
+end;
+
+{$ALIGN 8}
+type
+  LeapfrogHcState = record
+  private
+    hc1HivPop: PDouble;
+    hc1HivPopLength: Integer;
+    hc2HivPop: PDouble;
+    hc2HivPopLength: Integer;
+    hc1ArtPop: PDouble;
+    hc1ArtPopLength: Integer;
+    hc2ArtPop: PDouble;
+    hc2ArtPopLength: Integer;
+    hc1NoartAidsDeaths: PDouble;
+    hc1NoartAidsDeathsLength: Integer;
+    hc2NoartAidsDeaths: PDouble;
+    hc2NoartAidsDeathsLength: Integer;
+    hc1ArtAidsDeaths: PDouble;
+    hc1ArtAidsDeathsLength: Integer;
+    hc2ArtAidsDeaths: PDouble;
+    hc2ArtAidsDeathsLength: Integer;
+    hcArtInit: PDouble;
+    hcArtInitLength: Integer;
+    hcArtNeedInit: PDouble;
+    hcArtNeedInitLength: Integer;
+    hivBirths: PDouble;
+    hivBirthsLength: Integer;
+    ctxNeed: PDouble;
+    ctxNeedLength: Integer;
+    infectionByType: PDouble;
+    infectionByTypeLength: Integer;
+  public
+    procedure SetHc1HivPop(var inHc1HivPop: TGBFixedArray<Double>);
+    procedure SetHc2HivPop(var inHc2HivPop: TGBFixedArray<Double>);
+    procedure SetHc1ArtPop(var inHc1ArtPop: TGBFixedArray<Double>);
+    procedure SetHc2ArtPop(var inHc2ArtPop: TGBFixedArray<Double>);
+    procedure SetHc1NoartAidsDeaths(var inHc1NoartAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHc2NoartAidsDeaths(var inHc2NoartAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHc1ArtAidsDeaths(var inHc1ArtAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHc2ArtAidsDeaths(var inHc2ArtAidsDeaths: TGBFixedArray<Double>);
+    procedure SetHcArtInit(var inHcArtInit: TGBFixedArray<Double>);
+    procedure SetHcArtNeedInit(var inHcArtNeedInit: TGBFixedArray<Double>);
+    procedure SetHivBirths(var inHivBirths: TGBFixedArray<Double>);
+    procedure SetCtxNeed(var inCtxNeed: TGBFixedArray<Double>);
+    procedure SetInfectionByType(var inInfectionByType: TGBFixedArray<Double>);
+end;
+
+{$ALIGN 8}
+type
+  LeapfrogParams = record
+  private
+    dp: ^LeapfrogDpParams;
+    ha: ^LeapfrogHaParams;
```

Comment:
We have a `ha` module in Spectrum already so this might be a bit confusing. Can we use long names for these? "HivAdult", "HivChild", "DemProj"

Reply:
This was from Rob G, I've made this change now